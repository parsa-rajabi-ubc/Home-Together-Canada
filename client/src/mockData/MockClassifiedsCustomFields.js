/**
 * @Author:     Alex Qin
 * @Created:    2021.02.21
 *
 * @Description: mock data for custom fields.
 *
 */

export const rentalsMockCustomFields = {
    price: 800,
    numBed: 5,
    numBath: 3,
    furnished: true,
    petFriendly: false,
    smokingFriendly: false,
    pictures: "pictures"
}

export const houseServicesMockCustomFields = {
    rateAndFees: "666 per day",
    pictures: "pictures"
}

export const agenciesMockCustomFields = {
    rateAndFees: "555 per day",
    pictures: "pictures"
}

export const eventsMockCustomFields = {
    rateAndFees: "666 per day",
    eventDateTime: "2020/2/21 11:00 A.M.",
    contactName: "Alex",
    contactNumber:"778-555-5555",
    pictures: "pictures"
}