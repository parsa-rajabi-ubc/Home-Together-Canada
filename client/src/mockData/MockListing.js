/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.02.20
 *
 * @Description: Mock Listing
 *
 */
import {
    BUSINESS_SERVICE_CATEGORIES, MEMBER_SERVICE_CATEGORIES
} from "../createListing/constants/serviceListingCategoriesText";
import {BUSINESS_CLASSIFIEDS_CATEGORIES} from "../createListing/constants/classifiedListingCategoriesText";

var faker = require('faker');

export const mockServiceListings = [
    //CO-HOUSING 0
    {
        title: faker.lorem.words(),
        category: BUSINESS_SERVICE_CATEGORIES.CO_HOUSING,
        fullDescription: faker.lorem.paragraph(),
        contactName: faker.name.findName(),
        unitsForSale: faker.random.number(),
        unitsForRent: faker.random.number(),
    },
    // SHARED HOME 1
    {
        title: faker.lorem.words(),
        category: BUSINESS_SERVICE_CATEGORIES.SHARED_HOME_SERVICES,
        fullDescription: faker.lorem.paragraph(),
        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
    },
    // SHARED BUSINESS 2
    {
        title: faker.lorem.words(),
        category: BUSINESS_SERVICE_CATEGORIES.SHARED_BUSINESS_SERVICES,
        fullDescription: faker.lorem.paragraph(),
        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
    },
    // GOVERNMENT 3
    {
        title: faker.lorem.words(),
        category: BUSINESS_SERVICE_CATEGORIES.GOVERNMENT_SERVICES,
        fullDescription: faker.lorem.paragraph(),
        contactName: faker.name.findName(),
        contactPhoneNumber: faker.phone.phoneNumber(),
    },
    // MEMBER HOME TO SHARE 4
    {
        title: faker.lorem.words(),
        category: MEMBER_SERVICE_CATEGORIES.MEMBER_HOME,
        fullDescription: faker.lorem.paragraph(),

        generalLocation: faker.address.zipCode(),
        monthlyCost: faker.random.number(),
        pictures: faker.image.cats(),
        numBed: faker.random.number(),
        numBath: faker.random.number(),
        petFriendly: faker.random.boolean(),
        smokeFriendly: faker.random.boolean(),
        utilIncluded: faker.random.boolean(),
    }
]

export const mockClassifiedsListings = [
    //RENTALS 0
    {
        title: faker.lorem.words(),
        category: BUSINESS_CLASSIFIEDS_CATEGORIES.RENTALS,
        fullDescription: faker.lorem.paragraph(),

        price: faker.random.number(),
        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
        numBed: faker.random.number(),
        numBath: faker.random.number(),
        furnished: faker.random.boolean(),
        petFriendly: faker.random.boolean(),
        smokeFriendly: faker.random.boolean(),
    },
    // HOUSE_YARD 1
    {
        title: faker.lorem.words(),
        category: BUSINESS_CLASSIFIEDS_CATEGORIES.HOUSE_YARD,
        fullDescription: faker.lorem.paragraph(),

        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
    },
    // LEGAL_SALES 2
    {
        title: faker.lorem.words(),
        category: BUSINESS_CLASSIFIEDS_CATEGORIES.LEGAL_SALES,
        fullDescription: faker.lorem.paragraph(),

        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
    },
    // CLASSES_CLUBS 3
    {
        title: faker.lorem.words(),
        category: BUSINESS_CLASSIFIEDS_CATEGORIES.CLASSES_CLUBS,
        fullDescription: faker.lorem.paragraph(),

        contactName: faker.name.findName(),
        contactPhoneNumber: faker.phone.phoneNumber(),
        pictures: faker.image.cats(),
        ratesAndFees: faker.random.number() + " " + faker.random.words(),
        eventDateTime: faker.date.soon() + " " + faker.time.recent()
    }
]