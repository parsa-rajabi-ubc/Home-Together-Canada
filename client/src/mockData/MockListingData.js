/**
 * @Author:     Jeff Hatton
 * @Created:    2020.02.16
 *
 * @Description: Const array of mock listings - includes listings for members and businesses
 *
 */

export const mockMemberListings = [
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, Utilities Covered",
        monthlyCost: 750,
        petFriendly: false,
        smokeFriendly: false,
        datePosted: "2021-02-14T02:42:39.005Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Unfurnished Room, Utilities/internet Split",
        monthlyCost: 500,
        petFriendly: true,
        smokeFriendly: true,
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        title: "Basement suite",
        shortDescription: "1 bed 1 bath, nice part of town, parking spot",
        monthlyCost: 1100,
        petFriendly: false,
        smokeFriendly: false,
        datePosted: "2021-02-14T16:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, Utilities Covered",
        monthlyCost: 850,
        petFriendly: false,
        smokeFriendly: true,
        datePosted: "2021-02-16T23:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, Utilities Covered, Private Kitchenette",
        monthlyCost: 950,
        petFriendly: true,
        smokeFriendly: true,
        datePosted: "2021-02-16T03:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Unfurnished Room, spacious living area ",
        monthlyCost: 775,
        petFriendly: false,
        smokeFriendly: false,
        datePosted: "2021-02-15T11:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Unfurnished Room, large kitchen storage",
        monthlyCost: 750,
        petFriendly: true,
        smokeFriendly: false,
        datePosted: "2021-02-13T18:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, significant storage space",
        monthlyCost: 650,
        petFriendly: false,
        smokeFriendly: false,
        datePosted: "2021-02-13T05:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, private bathroom",
        monthlyCost: 750,
        petFriendly: true,
        smokeFriendly: false,
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, fast internet",
        monthlyCost: 850,
        petFriendly: false,
        smokeFriendly: false,
        datePosted: "2021-02-16T01:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished, private bath",
        monthlyCost: 950,
        petFriendly: true,
        smokeFriendly: false,
        datePosted: "2021-02-16T02:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, Utilities Covered",
        monthlyCost: 750,
        petFriendly: true,
        smokeFriendly: true,
        datePosted: "2021-02-14T08:42:39.000Z"
    },
    {
        title: "Rooms for Rent",
        shortDescription: "3 Furnished Rooms, for students preferably",
        monthlyCost: 650,
        petFriendly: false,
        smokeFriendly: true,
        datePosted: "2021-02-14T07:42:39.000Z"
    },
    {
        title: "Room for Rent",
        shortDescription: "Furnished Room, beautiful house",
        monthlyCost: 800,
        petFriendly: true,
        smokeFriendly: false,
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        title: "Basement suite for Couple",
        shortDescription: "Furnished Large Room, Utilities Covered",
        monthlyCost: 1300,
        petFriendly: true,
        smokeFriendly: true,
        datePosted: "2021-02-13T12:42:39.000Z"
    }
];

export const mockBusinessListings = [
    {
        logo: "public/uploads/logo1",
        title: "Apartments for 2 available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "2 bed 2 bath - furnished",
        datePosted: "2021-02-14T22:42:39.000Z"
    },
    {
        logo: "public/uploads/logo2",
        title: "Apartments for 3 available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "3 bed 2 bath - furnished",
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        logo: "public/uploads/logo3",
        title: "House with 4 rooms available",
        businessName: "Kelowna House Rentals",
        shortDescription: "4 bed 3 bath - furnished",
        datePosted: "2021-02-14T16:42:39.000Z"
    },
    {
        logo: "public/uploads/logo4",
        title: "Beautiful House On Lakefront For Rent",
        businessName: "Lakefront Properties",
        shortDescription: "2 bed 2 bath - furnished",
        datePosted: "2021-02-16T23:42:39.000Z"
    },
    {
        logo: "public/uploads/logo5",
        title: "Vacuum Rentals",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "2 bed 2 bath - furnished",
        datePosted: "2021-02-16T03:42:39.000Z"
    },
    {
        logo: "public/uploads/logo6",
        title: "Lawn Management Services",
        businessName: "Lawns-R-Us",
        shortDescription: "2 bed 2 bath - furnished",
        datePosted: "2021-02-15T11:42:39.000Z"
    },
    {
        logo: "public/uploads/logo7",
        title: "Carpet Cleaning Rentals",
        businessName: "Carpet Cleaners",
        shortDescription: "We offer services, or tool rentals",
        datePosted: "2021-02-13T18:42:39.000Z"
    },
    {
        logo: "public/uploads/logo8",
        title: "Plumbing Services Available",
        businessName: "Kelowna Plumbing Inc.",
        shortDescription: "2 bed 2 bath - furnished",
        datePosted: "2021-02-13T05:42:39.000Z"
    },
    {
        logo: "public/uploads/logo9",
        title: "Apartment Rooms Available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "Apartments of various sizes",
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        logo: "public/uploads/logo10",
        title: "Window Cleaning Services",
        businessName: "Kelowna Glass Cleaners",
        shortDescription: "We clean residential or high rise complexes",
        datePosted: "2021-02-16T01:42:39.000Z"
    },
    {
        logo: "public/uploads/logo11",
        title: "Apartments for 2 available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "2 bed 1 bath - unfurnished",
        datePosted: "2021-02-16T02:42:39.000Z"
    },
    {
        logo: "public/uploads/logo12",
        title: "Apartments for 2 available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "2 bed 2 bath - Unfurnished",
        datePosted: "2021-02-14T08:42:39.000Z"
    },
    {
        logo: "public/uploads/logo13",
        title: "Apartments for 3 available",
        businessName: "Kelowna Apartment Inc.",
        shortDescription: "3 bed 2 bath - Unfurnished",
        datePosted: "2021-02-14T07:42:39.000Z"
    },
    {
        logo: "public/uploads/logo14",
        title: "Senior services",
        businessName: "Senior services",
        shortDescription: "Mobility tools, awareness of area services, financial aid",
        datePosted: "2021-02-15T05:42:39.000Z"
    },
    {
        logo: "public/uploads/logo15",
        title: "Shared Vehicle Insurance",
        businessName: "Kelowna Vehicle Insurance",
        shortDescription: "Insurance and help available for groups sharing cars",
        datePosted: "2021-02-13T12:42:39.000Z"
    }
];