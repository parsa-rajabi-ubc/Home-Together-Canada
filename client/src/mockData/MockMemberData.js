/**
 * @Author:     Parsa Rajabi
 * @Created:    June 2022
 *
 * @Description: Mock Member data for automated testing
 *
 */
const {faker} = require("@faker-js/faker");

const MOCK_MEMBER_DATA = {
    firstName: faker.name.firstName(),
    lastName: faker.name.lastName(),
    email: 'tech@hometogether.ca',
    phoneNumber: {
        areaCode: faker.random.numeric(3),
        prefix: faker.random.numeric(3),
        suffix: faker.random.numeric(4)
    },
    address: {
        street: faker.address.streetAddress(),
        apt: faker.address.buildingNumber(),
        city: faker.address.city(),
        province: "Alberta",
        postCode: "T3T1T3"
    },
    monthlyRent: {
        min: Math.floor(Math.random() * 999) + 500,
        max: Math.floor(Math.random() * 2500) + 1000,
    },
    preferences:{
        petFriendly: faker.datatype.boolean(),
        smoking: faker.datatype.boolean(),
        mobile: faker.datatype.boolean(),
        allergies: faker.datatype.boolean(),
        religion: faker.datatype.boolean(),
        diet: faker.datatype.boolean(),
        hasHome: faker.datatype.boolean(),
        interestInBuyingHome: faker.datatype.boolean(),
    },
    aboutSelf: faker.random.words(2),

    sharingPreferences:{
        genders: ["Male", "Female", "Other"],
        age:{
            min: Math.floor(Math.random() * 29) + 18,
            max: Math.floor(Math.random() * 60) + 30,
        },
        overlapBudget:{
            min: Math.floor(Math.random() * 999) + 500,
            max: Math.floor(Math.random() * 2500) + 1000,
        }
    },
    accountDetails:{
        username: "TestAccount"+faker.random.numeric(2),
        password: "Bose1234"
    }

}

module.exports = {MOCK_MEMBER_DATA}