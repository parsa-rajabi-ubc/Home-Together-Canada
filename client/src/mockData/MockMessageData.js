/**
 * @Author:     Jeff Hatton
 * @Created:    2020.03.03
 *
 * @Description: Const array of mock messages - includes multiple conversations, most between messageMember and other members with differing timestamps
 *
 */

export const mockMessages = [
    {
        id: 1,
        messageContent: "message content 1 all on different days and times",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-01-12T02:42:39.005Z"
    },
    {
        id: 2,
        messageContent: "message content 2",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-01-01T02:42:39.005Z"
    },
    {
        id: 3,
        messageContent: "message content 3",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-02-14T08:42:39.005Z"
    },
    {
        id: 4,
        messageContent: "message content 4",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-01-12T02:42:39.005Z"
    },
    {
        id: 5,
        messageContent: "message content 5",
        senderId: 2,
        senderUsername: "member2",
        receiverId: 1,
        receiverUsername: "member1",
        dateSent: "2020-03-13T02:42:39.075Z"
    },
    {
        id: 6,
        messageContent: "message content 6",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-02-17T02:43:39.005Z"
    },
    {
        id: 7,
        messageContent: "message content 7",
        senderId: 2,
        senderUsername: "member2",
        receiverId: 1,
        receiverUsername: "member1",
        dateSent: "2021-01-19T09:42:39.005Z"
    },
    {
        id: 8,
        messageContent: "message content 8",
        senderId: 2,
        senderUsername: "member2",
        receiverId: 1,
        receiverUsername: "member1",
        dateSent: "2021-01-11T02:42:39.005Z"
    },
    {
        id: 9,
        messageContent: "message content 9 has timestamp before a smaller message id",
        senderId: 1,
        senderUsername: "member1",
        receiverId: 2,
        receiverUsername: "member2",
        dateSent: "2021-01-10T02:42:39.005Z"
    },
    {
        id: 10,
        messageContent: "message content 10",
        senderId: 2,
        senderUsername: "member2",
        receiverId: 1,
        receiverUsername: "member1",
        dateSent: "2021-02-10T02:42:39.005Z"
    }
];