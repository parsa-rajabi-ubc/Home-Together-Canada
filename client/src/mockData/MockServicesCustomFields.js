/**
 * @Author:     Alex Qin
 * @Created:    2021.02.18
 *
 * @Description: mock data for custom fields.
 *
 */


export const memberMockCustomFields = {
    streetLine1: "123 Candy Lane",
    streetLine2: "",
    city: "Kelowna",
    province: "BC",
    postalCode: "V1X 1X3",
    homeShareMonthlyCost: 800,
    numBed: 5,
    numBath: 3,
    utilIncluded: true,
    petFriendly: false,
    smokeFriendly: false,
    pictures: "pictures"
}

export const cohousingMockCustomFields = {
    contactName: "Alex",
    unitsForSale: 5,
    unitsForRent: 5,
}

export const homeServicesMockCustomFields = {
    rateAndFees: "666 per day",
    pictures: "pictures"
}

export const governmentMockCustomFields = {
    contactPerson: "Alex",
    phoneNumber:"778-555-5555"
}