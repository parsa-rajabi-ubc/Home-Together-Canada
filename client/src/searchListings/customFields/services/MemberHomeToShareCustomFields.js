/**
 * @Author:     Alex Qin
 * @Created:    2021.02.17
 *
 * @Description: Re-useable Member Home to share custom fields.
 *
 */
import PropTypes from "prop-types";
import React, {useState} from 'react';
import HTC_Logo from "../../../images/HTC_Logo.jpg";
import {resolveBooleanToYesNo} from "../../../common/utils/generalUtils";

const MemberHomeToShareCustomFields = (props) =>{
    const { reset } = props;
    const title = useState(reset.title);
    const generalLocationText = useState(reset.generalLocationText);
    const homeShareMonthlyCost = useState(reset.homeShareMonthlyCost);
    const numBed = useState(reset.numBed);
    const numBath = useState(reset.numBath);
    const utilIncluded = useState(resolveBooleanToYesNo(reset.utilIncluded));
    const petFriendly = useState(resolveBooleanToYesNo(reset.petFriendly));
    const smokeFriendly = useState(resolveBooleanToYesNo(reset.smokeFriendly));
    const photos = useState(reset.photos);

        return(
            <div>
                <h1 className={"page-title mb-5"}> {title} </h1> <br/>
                <p>General Location: {generalLocationText}</p> <br/>
                <p>Home share monthly cost: {homeShareMonthlyCost} ($CAD)</p> <br/>
                <p>Number of bedrooms available in home: {numBed}</p> <br/>
                <p>Number of bathrooms in house: {numBath}</p> <br/>
                <p>Utilities included: {utilIncluded}</p> <br/>
                <p>Pet friendly: {petFriendly}</p> <br/>
                <p>Smoking: {smokeFriendly}</p> <br/>
                <p>Photos: </p> <br/>
                {/*TODO: replace HTC_Logo with photos*/}
                <img className={"float-left w-24 h-auto mx-4 "} src={HTC_Logo} alt={""}/>
                <button className={"btn btn-green mb-6 w-1/2 text-base py-2"}>Giving Rate</button>
            </div>
        );
}

MemberHomeToShareCustomFields.propTypes = {
    reset: PropTypes.shape({
        title: PropTypes.string.isRequired,
        generalLocationText: PropTypes.string.isRequired,
        homeShareMonthlyCost: PropTypes.number.isRequired,
        numBed: PropTypes.string.isRequired,
        numBath: PropTypes.string.isRequired,
        utilIncluded: PropTypes.bool.isRequired,
        petFriendly: PropTypes.bool.isRequired,
        smokeFriendly: PropTypes.bool.isRequired,
        photos: PropTypes.string.isRequired,
    }).isRequired,
}

export default MemberHomeToShareCustomFields;