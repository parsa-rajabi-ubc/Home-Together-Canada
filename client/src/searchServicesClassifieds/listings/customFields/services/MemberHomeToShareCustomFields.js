/**
 * @Author:     Alex Qin
 * @Created:    2021.02.17
 *
 * @Description: Re-usable Member Home to share custom fields.
 *
 */
import PropTypes from "prop-types";
import React from 'react';
import {AiOutlineCheckCircle as Check, AiOutlineCloseCircle as Ex} from "react-icons/ai";
import {ImageGallery} from "../../../../common/listings/ImageGallery";

const MEMBER_HOME_TO_SHARE_CUSTOM_FIELDS_TEXT = {
    GENERAL_LOCATION: "General Location",
    HOME_SHARE_MONTHLY_COST: "Home Share Monthly Cost (CAD)",
    NUMBER_BEDROOM: "Number of Bedrooms",
    NUMBER_BATHROOM: "Number of Bathrooms",
    UTIL_INCLUDE: "Utilities included",
    PET_FRIENDLY: "Pet friendly",
    SMOKE_FRIENDLY: "Smoke friendly",
    PICTURES: "Pictures"
}

const MemberHomeToShareCustomFields = (props) => {
    const {
        streetLine1,
        streetLine2,
        city,
        province,
        postalCode,
        homeShareMonthlyCost,
        numBath,
        numBed,
        petFriendly,
        pictures,
        smokeFriendly,
        utilIncluded
    } = props;


    return (
        <div className={"flex"}>
            <div className={"flex-none w-9/12"}>
                <label className={"label-result"}>{MEMBER_HOME_TO_SHARE_CUSTOM_FIELDS_TEXT.GENERAL_LOCATION}</label>
                <p> {streetLine1} </p>
                <p> {streetLine2} </p>
                <p> {city}, {province} </p>
                <p> {postalCode} </p>

                <label
                    className={"label-result"}>{MEMBER_HOME_TO_SHARE_CUSTOM_FIELDS_TEXT.HOME_SHARE_MONTHLY_COST}</label>
                <p> {homeShareMonthlyCost}</p>

                <label className={"label-result"}>{MEMBER_HOME_TO_SHARE_CUSTOM_FIELDS_TEXT.NUMBER_BEDROOM}</label>
                <p> {numBed}</p>

                <label className={"label-result"}>{MEMBER_HOME_TO_SHARE_CUSTOM_FIELDS_TEXT.NUMBER_BATHROOM}</label>
                <p> {numBath}</p>

                {!!pictures.length &&
                    <div>
                        <label className={"label-result"}>{MEMBER_HOME_TO_SHARE_CUSTOM_FIELDS_TEXT.PICTURES}</label>
                        <ImageGallery pictures={pictures}/>
                    </div>
                }
            </div>


            <div className={"w-1/2"}>
                <table className={"table-auto mx-auto shadow-lg rounded-md"}>
                    <tbody>
                    <tr className={`${petFriendly && "bg-gray-200"}`}>
                        <td>
                            {petFriendly ?
                                <Check className={"checkmark-icon"}/> : <Ex className={"ex-icon"}/>
                            }
                        </td>
                        <td className={"table-item"}>{MEMBER_HOME_TO_SHARE_CUSTOM_FIELDS_TEXT.PET_FRIENDLY}</td>
                    </tr>

                    <tr className={`${smokeFriendly && "bg-gray-200"}`}>
                        <td>
                            {smokeFriendly ?
                                <Check className={"checkmark-icon"}/> : <Ex className={"ex-icon"}/>
                            }
                        </td>
                        <td className={"table-item"}>{MEMBER_HOME_TO_SHARE_CUSTOM_FIELDS_TEXT.SMOKE_FRIENDLY}</td>
                    </tr>

                    <tr className={`${utilIncluded && "bg-gray-200"}`}>
                        <td>
                            {utilIncluded ?
                                <Check className={"checkmark-icon"}/> : <Ex className={"ex-icon"}/>
                            }
                        </td>
                        <td className={"table-item"}>{MEMBER_HOME_TO_SHARE_CUSTOM_FIELDS_TEXT.UTIL_INCLUDE}</td>
                    </tr>

                    </tbody>
                </table>

                <button className={"btn btn-red my-10 text-base py-2"}>Report User
                </button>
            </div>


        </div>
    );
}

MemberHomeToShareCustomFields.propTypes = {
    streetLine1: PropTypes.string.isRequired,
    streetLine2: PropTypes.string.isRequired,
    city: PropTypes.string.isRequired,
    province: PropTypes.string.isRequired,
    postalCode: PropTypes.string.isRequired,
    homeShareMonthlyCost: PropTypes.number.isRequired,
    numBed: PropTypes.number.isRequired,
    numBath: PropTypes.number.isRequired,
    utilIncluded: PropTypes.bool.isRequired,
    petFriendly: PropTypes.bool.isRequired,
    smokeFriendly: PropTypes.bool.isRequired,
    pictures: PropTypes.string,
}

export default MemberHomeToShareCustomFields;