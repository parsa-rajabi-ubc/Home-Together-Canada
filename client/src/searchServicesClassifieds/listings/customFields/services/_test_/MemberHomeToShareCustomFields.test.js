/**
 * @Author:     Alex Qin
 * @Created:    2021.02.17
 *
 * @Description: MemberHomeToShareCustomFields snapshot test. Returns a snapshot.
 *
 */
import React from 'react';
import renderer from  'react-test-renderer'
import {BrowserRouter as Router} from "react-router-dom";
import {Member} from "../../../../../mockData/MockServicesCustomFields";
import MemberHomeToShareCustomFields from "../MemberHomeToShareCustomFields";

describe('MemberHomeToShareCustomFields', () => {
    it("should render correctly regardless of properties", () => {
        // given
        const props = {
            ...Member,
        };
        //when
        const component = renderer.create(<Router><MemberHomeToShareCustomFields {...props}/></Router>);
        const tree = component.toJSON();
        //then
        expect(tree).toMatchSnapshot();
    });
});