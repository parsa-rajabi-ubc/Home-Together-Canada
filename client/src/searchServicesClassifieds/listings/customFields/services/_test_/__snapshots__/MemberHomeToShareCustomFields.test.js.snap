// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberHomeToShareCustomFields should render correctly regardless of properties 1`] = `
<div>
  <label
    className="label-result"
  >
    General Location
  </label>
  <p>
     
    766 Rutland Rd. S.
  </p>
  <label
    className="label-result"
  >
    Home Share Monthly Cost (CAD)
  </label>
  <p>
     
    800
  </p>
  <label
    className="label-result"
  >
    Number of Bedrooms
  </label>
  <p>
     
    5
  </p>
  <label
    className="label-result"
  >
    Number of Bedrooms
  </label>
  <p>
     
    5
  </p>
  <label
    className="label-result"
  >
    Number of Bathrooms
  </label>
  <p>
     
    3
  </p>
  <label
    className="label-result"
  >
    Utilities included
  </label>
  <p>
     
  </p>
  <label
    className="label-result"
  >
    Pet friendly
  </label>
  <p>
     
  </p>
  <label
    className="label-result"
  >
    Smoke friendly
  </label>
  <p>
     
  </p>
  <label
    className="label-result"
  />
  <img
    alt=""
    src="HTC_Logo.jpg"
  />
</div>
`;
